onebusaway-android/src/main/java/org/onebusaway/android/ui/HomeActivity.java

```diff
@@ -1997,14 +1997,25 @@ private void initWeatherView(){
         weatherView = findViewById(R.id.weatherView);
     }
 
-    private void setWeatherData(){
+    private void setWeatherData() {
         TextView tempTxtView = findViewById(R.id.weatherTextView);
         ImageView weatherImageView = findViewById(R.id.weatherStateImageView);
+        String weatherIcon = weatherResponse.getCurrent_forecast().getIcon();
```

Comment:
what happens if `getCurrent_forecast()` returns `null`?

Reply:
`getCurrent_forecast()` will not be null I have added validation on line `2040`

`weatherResponse.getCurrent_forecast() != null`

Reply:
I can't reproduce the issue but I have added logical changes to avoid this happen